From a5867da0031021aed4639cbb56f1ca4861e8f855 Mon Sep 17 00:00:00 2001
From: Pavel Zhukov <pavel@zhukoff.net>
Date: Thu, 13 Apr 2023 14:56:14 +0200
Subject: [PATCH 2/2] Check if fdt_addr is defined
Cc: pavel@zhukoff.net

---
 scripts/uboot-script-gen | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/scripts/uboot-script-gen b/scripts/uboot-script-gen
index 765723a..db5f747 100755
--- a/scripts/uboot-script-gen
+++ b/scripts/uboot-script-gen
@@ -409,7 +409,8 @@ function device_tree_editing()
 
     if test $UBOOT_SOURCE
     then
-        echo "fdt addr $device_tree_addr" >> $UBOOT_SOURCE
+	echo 'test -n "${fdt_addr}" && setenv device_tree_addr ${fdt_addr} || sentenv device_tree_addr' "${device_tree_addr}" >> $UBOOT_SOURCE
+	echo 'fdt addr ${device_tree_addr}' >> $UBOOT_SOURCE
         echo "fdt resize 1024" >> $UBOOT_SOURCE
 
         if test $NUM_DT_OVERLAY && test $NUM_DT_OVERLAY -gt 0
@@ -1312,7 +1313,7 @@ fi
 
 check_file_type $DEVICE_TREE "Device Tree Blob"
 device_tree_addr=$memaddr
-load_file $DEVICE_TREE "host_fdt"
+test -n "$DEVICE_TREE" && load_file $DEVICE_TREE "host_fdt"
 bitstream_load_and_config  # bitstream is loaded last but used first
 device_tree_editing $device_tree_addr
 
@@ -1329,7 +1330,7 @@ fi
 
 if [ "$BOOT_CMD" != "none" ]
 then
-    echo "$BOOT_CMD $kernel_addr - $device_tree_addr" >> $UBOOT_SOURCE
+    echo "$BOOT_CMD $kernel_addr" '- ${device_tree_addr}' >> $UBOOT_SOURCE
 else
     # skip boot command but store load addresses to be used later
     echo "setenv host_kernel_addr $kernel_addr" >> $UBOOT_SOURCE
-- 
2.39.2

