diff --git a/samples/hello_world/src/main.c b/samples/hello_world/src/main.c
index 9a90c6a6a0..47d73bd6e7 100644
--- a/zephyr/samples/hello_world/src/main.c
+++ b/zephyr/samples/hello_world/src/main.c
@@ -4,9 +4,25 @@
  * SPDX-License-Identifier: Apache-2.0
  */

+#include "zephyr/xen/events.h"
+#define SHAREDMEM_BASE 0x6000000
+#define SLEEPTIME 5000
 #include <zephyr/kernel.h>
 
 void main(void)
 {
+       printk("Main starting!\n");
+	int *ptr;
+	ptr = SHAREDMEM_BASE;
+       printk("Pointer ready to use!\n");
+	*ptr = 0;
 	printk("Hello World! %s\n", CONFIG_BOARD);
+	printk("Memory initialized: %p => %d !\n", ptr, *ptr);
+	k_msleep(SLEEPTIME*10);
+	while (1){
+		printk("%s: Hello automotive team! Shared memory %p value: %d \n",
+				CONFIG_BOARD, ptr, *ptr);
+		(*ptr)++;
+		k_msleep(SLEEPTIME);
+	}
 }
diff --git a/soc/arm64/xenvm/mmu_regions.c b/soc/arm64/xenvm/mmu_regions.c
index 43e3ada82e..0fe57e49cd 100644
--- a/zephyr/soc/arm64/xenvm/mmu_regions.c
+++ b/zephyr/soc/arm64/xenvm/mmu_regions.c
@@ -23,6 +23,12 @@ static const struct arm_mmu_region mmu_regions[] = {
 			      DT_REG_ADDR_BY_IDX(DT_INST(0, xen_xen), 0),
 			      DT_REG_SIZE_BY_IDX(DT_INST(0, xen_xen), 0),
 			      MT_NORMAL | MT_P_RW_U_NA | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("HYPERCAR_MEM_O",
+			      0x6000000,
+			      0x1000000,
+			      MT_NORMAL_NC | MT_P_RW_U_RW | MT_NS),
+
 };
 
 const struct arm_mmu_config mmu_config = {
