diff --git a/samples/hello_world/src/main.c b/samples/hello_world/src/main.c
index 9a90c6a6a0..47d73bd6e7 100644
--- a/zephyr/samples/hello_world/src/main.c
+++ b/zephyr/samples/hello_world/src/main.c
@@ -4,9 +4,37 @@
  * SPDX-License-Identifier: Apache-2.0
  */

+#include "zephyr/xen/events.h"
+#define SHAREDMEM_BASE 0x6000000
+#define SLEEPTIME 500
+#define DOMNUM 10
 #include <zephyr/kernel.h>
 
 void main(void)
 {
+       printk("Main starting!\n");
+	int (*ptr)[DOMNUM];
+       int last[DOMNUM];
+	ptr = SHAREDMEM_BASE;
+       printk("Pointer ready to use!\n");
+	for (int i=0; i<DOMNUM-3; i++){
+	    (*ptr)[i]=0;
+           last[i] = 0;
+	}
 	printk("Hello World! %s\n", CONFIG_BOARD);
+	printk("Memory initialized: %p => %d !\n", ptr, *ptr);
+	while (1){
+               for (int i=0; i<DOMNUM; i++){
+		    int current = (*ptr)[i];
+		    int prev = last[i];
+		    if (prev == 0 && current != 0){
+		       printk("RTOS: Domain %d is alive! \n", i);
+		    };
+		    if (prev != current && prev + 1 != current){
+		       printk(" RTOS: ALERT! ALERT! ALERT! DomU%d misbehave, restart it!  \n ", i, prev, current);
+		    };
+		    last[i] = current;
+		}
+		k_msleep(SLEEPTIME);
+	}
 }
