diff --git a/boards/arm64/xenvm/xenvm_gicv3_defconfig b/boards/arm64/xenvm/xenvm_gicv3_defconfig
index b5b63987b49..af7e2c823c9 100644
--- a/zephyr/boards/arm64/xenvm/xenvm_gicv3_defconfig
+++ b/zephyr/boards/arm64/xenvm/xenvm_gicv3_defconfig
@@ -18,3 +18,5 @@ CONFIG_LOG=y
 CONFIG_LOG_MODE_MINIMAL=n
 
 CONFIG_USERSPACE=n
+CONFIG_XEN_INITIAL_DOMAIN=y
+CONFIG_UART_XEN_HVC_CONSOLEIO=y
diff --git a/boards/arm64/xenvm/xenvm.dts b/boards/arm64/xenvm/xenvm.dts
index 00607823886..a0b3bbad121 100644
--- a/zephyr/boards/arm64/xenvm/xenvm.dts
+++ b/zephyr/boards/arm64/xenvm/xenvm.dts
@@ -26,7 +26,7 @@
 
 	chosen {
 		zephyr,sram = &ram;
-		zephyr,console = &xen_hvc;
+		zephyr,console = &xen_consoleio_hvc;
 	};
 
 	cpus {
@@ -74,7 +74,7 @@
 		interrupt-parent = <&gic>;
 	};
 
-	xen_hvc: hvc {
+	xen_consoleio_hvc: hvc {
 		compatible = "xen,hvc-uart";
 		status = "okay";
 	};
diff --git a/drivers/serial/uart_hvc_xen.c b/drivers/serial/uart_hvc_xen.c
index 478a1e0b399..cab2b88b6a3 100644
--- a/zephyr/drivers/serial/uart_hvc_xen.c
+++ b/zephyr/drivers/serial/uart_hvc_xen.c
@@ -262,7 +262,7 @@ int xen_console_init(const struct device *dev)
 	return 0;
 }
 
-DEVICE_DT_DEFINE(DT_NODELABEL(xen_hvc), xen_console_init, NULL, &hvc_data,
+DEVICE_DT_DEFINE(DT_NODELABEL(xen_consoleio_hvc), xen_console_init, NULL, &hvc_data,
 		NULL, PRE_KERNEL_1, CONFIG_XEN_HVC_INIT_PRIORITY,
 		&xen_hvc_api);
 
diff --git a/boards/arm64/xenvm/xenvm_defconfig b/boards/arm64/xenvm/xenvm_defconfig
index 8905b74a0c9..4eec4bdb982 100644
--- a/zephyr/boards/arm64/xenvm/xenvm_defconfig
+++ b/zephyr/boards/arm64/xenvm/xenvm_defconfig
@@ -16,3 +16,5 @@ CONFIG_LOG=y
 CONFIG_LOG_MODE_MINIMAL=n
 
 CONFIG_USERSPACE=n
+CONFIG_UART_XEN_HVC_CONSOLEIO=y
+CONFIG_XEN_INITIAL_DOMAIN=y
\ No newline at end of file
